### PR TITLE
obs-filters: Add NVIDIA Blur Filter & Background Blur

### PR DESCRIPTION
**Pull request by pkviet**
### Description
This adds a Blur filter based on NVIDIA Video Effects, leveraging its Background Blur Filter.
The hack is to set the whole frame as background.

**Update from August 5, 2024:**
This adds the Background Blur filter.
See https://github.com/obsproject/obs-studio/pull/9533#issuecomment-2267848949

### Motivation and Context
Seems there's interest in having a gaussian blur filter.
Some use cases:
- a "brb" screen that has the blurred game screen in the background. (Rodney)
- blurring my camera and capture sources for BRB scenes and such (Mitchell)
- blurring in game chat, cus people say sensitive/bad stuff, and looks better than me just slapping an img on top of it (Flaeri)

There are several third-party blur filters; it'd be nice to have a native one.

**Update from August 5, 2024:**
The background blur filter was added due to its usefulness for presenters.

### How Has This Been Tested?
Tested the blur on several sources with an RTX 4090 with both SDR / HDR content on SDR & HDR monitor:
- image source;
- media source;
- window-capture (on HDR monitor).

obs64_2023-09-03_23-37-26

obs64_2023-08-28_23-43-11

GPU usage seems negligible.

### Types of changes
 - New feature (non-breaking change which adds functionality) 

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by pkviet** [comment]
Rodney has run the filter in his profiler and it seems fine. It's generally fast (2 ms) with occasional spikes due to his GPU reclocking.
2023-08-29_00-16-27_WefKA0

**Comment by pkviet** [comment]
The testers observe that the filter brightens the source; I can confirm it's the case.

**Comment by exeldro** [comment]
I would suggest putting all NVIDIA Video Effects in a single filter and not making different filters for each. So that we support all these:
- AI green screen filter
- Background Blur filter
- AI-based artifact reduction
- AI-based super resolution
- AI-based fast video upscaler
- Artifact removal filter
- Super resolution filter
From https://docs.nvidia.com/deeplearning/maxine/vfx-sdk-programming-guide/index.html#nvvfx-effectselector

**Comment by pkviet** [comment]
> I would suggest putting all NVIDIA Video Effects in a single filter and not making different filters for each. So that we support all these:
> 
> * AI green screen filter
> * Background Blur filter
> * AI-based artifact reduction
> * AI-based super resolution
> * AI-based fast video upscaler
> * Artifact removal filter
> * Super resolution filter
> 
> From https://docs.nvidia.com/deeplearning/maxine/vfx-sdk-programming-guide/index.html#nvvfx-effectselector

There's 0 plan to add any of these effects, at least on my side.
I prefer to split the effects (background removal & blur) in different filters because the swapping of effects is shitty in my experience. So I prefer the low hanging fruit.
That said, the filter is buggy atm; it's too bright, so it's not close to being merged.

**Comment by norihiro** [comment]
The commit title should be in present tense. I'd like to suggest `Add NVIDIA blur filter`.

**Comment by pkviet** [comment]
**Update**
- Fixed the brightness issue.
- Tested on HDR content; seems to work.

**Comment by pkviet** [comment]
> The commit title should be in present tense. I'd like to suggest `Add NVIDIA blur filter`.

Agreed; fixed it

**Comment by pkviet** [comment]
**Update (August 5, 2024)**
- This PR requires now PR #9457 which factors out from obs-filters the NVIDIA Audio/Video FX.
- This update adds the Background Blur filter to the Blur filter. This is useful for speakers.
- (right: original; left: blurred background)
obs64_2024-08-05_00-48-50

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 1]
```diff
@@ -149,73 +216,252 @@ static void nv_greenscreen_filter_actual_destroy(void *data)
...
+		nvvfx_new_sdk = filter->version >= MIN_VFX_SDK_VERSION &&
+				nvvfx_new_sdk;
+	}
+#endif
+	/* 1. Create FX */
+	/* 2. Set models path & initialize CudaStream */
+	if (!nvvfx_filter_create_internal(filter))
+		return NULL;
+
+	/* 3. Load effect. */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:343]:
```suggestion
	/* 3. Load effect */
```

Generally "bullet points" don't end in final punctuation, unless the bullet point text requires non-trivial punctuation (multiple sentences, or other punctuation).

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 2]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
+	return true;
+}
+
+static bool create_blur_texture(struct nvvfx_data *filter)
+{
+	NvCV_Status vfxErr;
+	uint32_t width = filter->width;
+	uint32_t height = filter->height;
+
+	/* 1. create blur texture */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:505]:
```suggestion
	/* 1. Create blur texture */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 3]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
+		gs_texture_create(width, height, GS_RGBA_UNORM, 1, NULL, 0);
+	if (filter->blur_texture == NULL) {
+		error("Blur texture couldn't be created");
+		return false;
+	}
+	struct ID3D11Texture2D *d11texture =
+		(struct ID3D11Texture2D *)gs_texture_get_obj(
+			filter->blur_texture);
+
+	/* 2. Create NvCVImage which will hold final blur texture. */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:519]:
```suggestion
	/* 2. Create NvCVImage which will hold final blur texture */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 4]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
-		}
+	return true;
+}
+
+static bool init_blur_images(struct nvvfx_data *filter)
+{
+	uint32_t width = filter->width;
+	uint32_t height = filter->height;
+
+	/* 1. Create and allocate Blur BGR NvCVimage (blur fx dst). */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:561]:
```suggestion
	/* 1. Create and allocate Blur BGR NvCVimage (blur fx dst) */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 5]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
-		if (NvCVImage_Create(width, height, NVCV_BGR, NVCV_U8,
-				     NVCV_CHUNKY, NVCV_GPU, 1,
-				     &filter->BGR_src_img) != NVCV_SUCCESS) {
+		NvCVImage_Create(width, height, NVCV_RGBA, NVCV_U8, NVCV_CHUNKY,
+				 NVCV_GPU, 1, &filter->RGBA_dst_img);
+		NvCVImage_Alloc(filter->RGBA_dst_img, width, height, NVCV_RGBA,
+				NVCV_U8, NVCV_CHUNKY, NVCV_GPU, 1);
+	}
+
+	/* 3. Set input & output images for nv blur FX. */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:578]:
```suggestion
	/* 3. Set input & output images for NV blur FX */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 6]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
+	return true;
+}
+
+static void init_images(struct nvvfx_data *filter)
+{
+	NvCV_Status vfxErr;
+	uint32_t width = filter->width;
+	uint32_t height = filter->height;
+
+	/* 1. create alpha texture & associated NvCVImage */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:600]:
```suggestion
	/* 1. Create alpha texture & associated NvCVImage */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 7]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
+	if (filter->filter_id == S_FX_BG_BLUR ||
+	    filter->filter_id == S_FX_AIGS) {
+		if (!create_alpha_texture(filter))
 			goto fail;
-		}
-		if (NvCVImage_Alloc(filter->BGR_src_img, width, height,
-				    NVCV_BGR, NVCV_U8, NVCV_CHUNKY, NVCV_GPU,
-				    1) != NVCV_SUCCESS) {
+	}
+	/* 2. create blur texture & associated NvCVImage */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:606]:
```suggestion
	/* 2. Create blur texture & associated NvCVImage */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 8]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
+	if (filter->filter_id == S_FX_BG_BLUR ||
+	    filter->filter_id == S_FX_BLUR) {
+		if (!create_blur_texture(filter))
 			goto fail;
-		}
 	}
+	/* 3. create texrenders */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:612]:
```suggestion
	/* 3. Create texrenders */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 9]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
+		goto fail;
 
-	/* 5. Create and allocate Alpha NvCVimage (fx dst). */
-	if (filter->A_dst_img) {
-		if (NvCVImage_Realloc(filter->A_dst_img, width, height, NVCV_A,
-				      NVCV_U8, NVCV_CHUNKY, NVCV_GPU,
-				      1) != NVCV_SUCCESS) {
-			goto fail;
-		}
+	/* 4. Create and allocate BGR NvCVImage (alpha mask fx src). */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:616]:
```suggestion
	/* 4. Create and allocate BGR NvCVImage (alpha mask fx src) */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 10]
```diff
@@ -229,114 +475,199 @@ static void init_images_greenscreen(struct nv_greenscreen_data *filter)
...
 	}
 
-	/* 6. Create stage NvCVImage which will be used as buffer for transfer */
-	if (filter->stage) {
-		if (NvCVImage_Realloc(filter->stage, width, height, NVCV_RGBA,
-				      NVCV_U8, NVCV_PLANAR, NVCV_GPU,
-				      1) != NVCV_SUCCESS) {
-			goto fail;
-		}
+	/* 5. Create and allocate Alpha NvCVimage (mask fx dst). */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:627]:
```suggestion
	/* 5. Create and allocate Alpha NvCVimage (mask fx dst) */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 11]
```diff
@@ -359,229 +707,138 @@ static bool process_texture_greenscreen(struct nv_greenscreen_data *filter)
...
-	if (filter->effect) {
-		filter->mask_param =
-			gs_effect_get_param_by_name(filter->effect, "mask");
-		filter->image_param =
-			gs_effect_get_param_by_name(filter->effect, "image");
-		filter->threshold_param = gs_effect_get_param_by_name(
-			filter->effect, "threshold");
-		filter->multiplier_param = gs_effect_get_param_by_name(
-			filter->effect, "multiplier");
+	/* 3. run AIGS fx */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:725]:
```suggestion
	/* 3. Run AIGS FX */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 12]
```diff
@@ -359,229 +707,138 @@ static bool process_texture_greenscreen(struct nv_greenscreen_data *filter)
...
 	}
-	obs_leave_graphics();
 
-	/* 4. Allocate state for the effect */
-	if (nvvfx_new_sdk) {
-		vfxErr = NvVFX_AllocateState(filter->handle,
-					     &filter->stateObjectHandle);
-		if (NVCV_SUCCESS != vfxErr) {
+	if (id != S_FX_AIGS) {
+		/* 4. BLUR fx */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:739]:
```suggestion
		/* 4. Blur FX */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 13]
```diff
@@ -359,229 +707,138 @@ static bool process_texture_greenscreen(struct nv_greenscreen_data *filter)
...
+		/* 4a. run BLUR fx except for AIGS */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:740]:
```suggestion
		/* 4a. Run BLUR FX except for AIGS */
```

**Review by RytoEX — CHANGES_REQUESTED** [review]
Did a quick visual pass. Would appreciate further review and testing, as I do not have an appropriate system currently set up for testing this.

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 14]
```diff
@@ -50,25 +56,29 @@ struct nv_greenscreen_data {
 	bool got_new_frame;
 	signal_handler_t *handler;
 
+	bool uses_greenscreen;
+	bool uses_blur;
```
[the remark this answers is not in the document]
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:60]: The 2 bools are actually a remnant from earlier versions of the PR and are not used. I had forgotten to remove them... 
But for the sake of completeness, the answer to the question is: yes it's possible; the background blur of nvidia SDK runs two fx (greenscreen fx + blur fx proper); i hacked the sdk to be able to use blur fx without greenscreen fx to create the obs blur filter, distinguished from obs background blur fiter. The bools were tracking the nvidia fx being used, but i opted for another mechanics eventually to track what's needed to be run.

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 15]
```diff
@@ -50,25 +56,29 @@ struct nv_greenscreen_data {
...
 	NvVFX_Handle handle;
 	CUstream stream;        // CUDA stream
 	int mode;               // 0 = quality, 1 = performance
 	NvCVImage *src_img;     // src img in obs format (RGBA ?) on GPU
 	NvCVImage *BGR_src_img; // src img in BGR on GPU
 	NvCVImage *A_dst_img;   // mask img on GPU
-	NvCVImage *dst_img;     // mask texture
-	NvCVImage *stage;       // planar stage img used for transfer to texture
+	NvCVImage *
+		dst_img; // AIGS: alpha mask texture; blur: texture initialized from d3d11 (RGBA, chunky, u8)
```
[the remark this answers is not in the document]
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:70]: thanks, good idea; done

[the remark this answers is not in the document]
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:70]: it's what's defined in nvidia sdk, AI Greenscreen. I've replaced by 'greenscreen' to reflect the naming used so far in obs

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 16]
```diff
@@ -79,9 +89,22 @@ struct nv_greenscreen_data {
...
+
+	/* blur specific */
+	enum nvvfx_filter_id filter_id;
+	NvVFX_Handle handle_blur;
+	CUstream stream_blur;        // cuda stream
+	float strength;              // from 0 to 1, default = 0.5
+	NvCVImage *blur_BGR_dst_img; // dst img of blur fx (BGR, chunky, u8)
+	NvCVImage *RGBA_dst_img;     // tmp img used to transfer to texture
+	NvCVImage *
+		blur_dst_img; // dst img initialized from d3d11 texture (RGBA, chunky, u8)
```
[the remark this answers is not in the document]
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:105]: done

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 9 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:616]: done

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 10 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:627]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 11 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:725]: fixed and I added the explanation for AIGS = AI Greenscreen

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 13 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:740]: done

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 12 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c]: done

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 17]
```diff
@@ -90,26 +113,57 @@ static const char *nv_greenscreen_filter_name(void *unused)
...
+			filter->mode = mode;
+			vfxErr = NvVFX_SetU32(filter->handle, NVVFX_MODE, mode);
+			vfxErr = NvVFX_Load(filter->handle);
+			if (NVCV_SUCCESS != vfxErr)
+				error("Error loading AI Greenscreen FX %i",
+				      vfxErr);
+		}
+	}
+	if (id == S_FX_BLUR || id == S_FX_BG_BLUR) {
+
```
[the remark this answers is not in the document]
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:151]: oh, right

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 18]
```diff
@@ -119,7 +173,10 @@ static void nv_greenscreen_filter_actual_destroy(void *data)
 
 	if (filter->images_allocated) {
 		obs_enter_graphics();
-		gs_texture_destroy(filter->alpha_texture);
+		if (filter->filter_id == S_FX_AIGS)
+			gs_texture_destroy(filter->alpha_texture);
+		else
+			gs_texture_destroy(filter->blur_texture);
```
[the remark this answers is not in the document]
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:178]: done

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 1 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:342]: thanks, point noted

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 2 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:504]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 3 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:518]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 4 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:560]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 5 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:577]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 6 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:599]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 7 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:605]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 8 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:611]: fixed

File: plugins/nv-filters/data/rtx_blur.effect [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+
+struct VertData {
+	float4 pos : POSITION;
+	float2 uv  : TEXCOORD0;
+};
+
+struct VertInOut {
+	float2 uv  : TEXCOORD0;
+	float4 pos : POSITION;
+};
```
[the remark this answers is not in the document]
Comment (RytoEX) [reply plugins/nv-filters/data/rtx_blur.effect:23]: Good enough for me.

**Comment by pkviet** [comment]
@derrod @flaeri @prgmitchell @WizardCM would appreciate if you could test/retest the PR since Ryan doesn't have easy access to RTX GPU.

**Review by derrod — APPROVED** [review]
LGTM. Having native background blur is nice. The blur itself not being vendor-agnostic is a bit weird but at least we'll have it for the majority of users (i.e. Windows + NVIDIA systems).

**Comment by prgmitchell** [comment]
> @derrod @flaeri @prgmitchell @WizardCM would appreciate if you could test/retest the PR since Ryan doesn't have easy access to RTX GPU.

same as Rodney, working fine with pretty minimal performance impact on my 3080.

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 20]
```diff
@@ -57,18 +63,20 @@ struct nv_greenscreen_data {
 	NvCVImage *src_img;     // src img in obs format (RGBA ?) on GPU
 	NvCVImage *BGR_src_img; // src img in BGR on GPU
 	NvCVImage *A_dst_img;   // mask img on GPU
-	NvCVImage *dst_img;     // mask texture
-	NvCVImage *stage;       // planar stage img used for transfer to texture
+	/* clang-format off */
+	NvCVImage *dst_img; // Greenscreen: alpha mask texture; blur: texture initialized from d3d11 (RGBA, chunky, u8)
+	/* clang-format on */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:68]: I meant you can just use `/* clang-format off */` and `/* clang-format on */` for the entire struct, but align the comments manually. Ignore the column length limit.

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 21]
```diff
@@ -38,9 +37,16 @@
 #define TEXT_PROCESSING MT_("Nvvfx.Method.Greenscreen.Processing")
 #define TEXT_PROCESSING_HINT MT_("Nvvfx.Method.Greenscreen.Processing.Hint")
 
+/* blur & background blur fx */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:40]:
```suggestion
/* Blur & Background Blur FX */
```

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 22]
```diff
@@ -79,9 +87,23 @@ struct nv_greenscreen_data {
...
+	/* blur specific */
+	enum nvvfx_filter_id filter_id;
+	NvVFX_Handle handle_blur;
+	CUstream stream_blur;        // cuda stream
+	float strength;              // from 0 to 1, default = 0.5
+	NvCVImage *blur_BGR_dst_img; // dst img of blur fx (BGR, chunky, u8)
+	NvCVImage *RGBA_dst_img;     // tmp img used to transfer to texture
+	/* clang-format off */
+	NvCVImage *blur_dst_img; // dst img initialized from d3d11 texture (RGBA, chunky, u8)
+	/* clang-format on */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:104]: Same here. You can turn clang-format off for the whole struct, but align the comments manually.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Final nits.

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 21 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 20 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 22 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c]: fixed

File: plugins/nv-filters/nvidia-videofx-filter.c [hunk 23]
```diff
@@ -38,9 +37,17 @@
...
+#define S_STRENGTH "intensity"
+#define S_STRENGTH_DEFAULT 0.5
+#define TEXT_MODE_BLUR_STRENGTH MT_("Nvvfx.Method.Blur.Strength")
+
+enum nvvfx_filter_id { S_FX_AIGS, S_FX_BLUR, S_FX_BG_BLUR };
+
 bool nvvfx_loaded = false;
 bool nvvfx_new_sdk = false;
-struct nv_greenscreen_data {
+/* clang-format off */
```
Comment (RytoEX) [inline plugins/nv-filters/nvidia-videofx-filter.c:49]: Put a blank line above this.

**Review by RytoEX — CHANGES_REQUESTED** [review]
One last nit, then I'm good.

File: plugins/nv-filters/nvidia-videofx-filter.c (hunk 23 above)
Comment (pkviet) [reply plugins/nv-filters/nvidia-videofx-filter.c:50]: done

**Review by RytoEX — APPROVED** [review]
Once CI passes, this is good enough for me.

**Pull request merged**